contracts/GenArt721CoreV3.sol

```diff
@@ -162,30 +160,31 @@ contract GenArt721CoreV3 is ERC721Enumerable, IGenArt721CoreContractV3 {
         internal
         returns (uint256 _tokenId)
     {
-        uint256 tokenIdToBe = (_projectId * ONE_MILLION) +
+        uint256 nextTokenId = (_projectId * ONE_MILLION) +
             projects[_projectId].invocations;
 
         projects[_projectId].invocations = projects[_projectId].invocations + 1;
 
-        bytes32 hash = keccak256(
+        bytes32 tokenHash = keccak256(
             abi.encodePacked(
                 projects[_projectId].invocations,
                 block.number,
                 blockhash(block.number - 1),
                 randomizerContract.returnValue()
             )
         );
-        tokenIdToHash[tokenIdToBe] = hash;
-        hashToTokenId[hash] = tokenIdToBe;
 
-        _mint(_to, tokenIdToBe);
+        tokenIdToHash[nextTokenId] = tokenHash;
+        hashToTokenId[tokenHash] = nextTokenId;
 
-        tokenIdToProjectId[tokenIdToBe] = _projectId;
-        //projectIdToTokenIds[_projectId].push(tokenIdToBe);
+        _mint(_to, nextTokenId);
 
-        emit Mint(_to, tokenIdToBe, _projectId);
+        // Do not need to also log `projectId` in event, as the `projectId` for
+        // a given token can be derived from the `tokenId` with:
+        //   projectId = tokenId - (tokenId % 1_000_000)
+        emit MintToken(_to, nextTokenId);
```

Comment:
I'm good with this. We should just be aware that with this pattern, w can no longer filter mint events to a specific project when using web3 calls. This is because events use bloom filter logs, so only searches for specific values are possible. In reality, we would rely on the subgraph for this functionality, and I think this makes total sense!

Reply:
Any reason we shouldn't just call the new event `Mint`?

Reply:
Roger that–that makes sense to me and seems like an OK trade off given that we use the subgraph anyway for reading indexed data. Only being able to filter with web3 calls for tokenID or minter address seems OK to me for the mint events themselves.

> Any reason we shouldn't just call the new event Mint?
No strong opinion here at all. Figured we may just want to name differently than old `Mint` event to make it clearer that a change was made.

Reply:
>Figured we may just want to name differently than old Mint event to make it clearer that a change was made.

I think rolling the contract version and changing the qty of args is enough to distinguish the old Mint from the new Mint 🤷 no super strong opinions other than MintToken seems a bit verbose.

Reply:
Yeah that's true + fair enough. I have ~0 opinion here, so happy to switch back to `Mint` to avoid verbosity.